cmake/AwsThreadName.cmake

```diff
@@ -52,7 +54,18 @@ function(aws_set_thread_name_method target)
         " PTHREAD_SETNAME_TAKES_3ARGS)
     if (PTHREAD_SETNAME_TAKES_3ARGS)
         target_compile_definitions(${target} PRIVATE -DAWS_PTHREAD_SETNAME_TAKES_3ARGS)
-        return()
+        #return()
+    endif()
+
+    check_c_source_compiles("
+        ${c_source_start}
+        char name[16] = {0};
+        pthread_getname_np(thread_id, name, 16);
+        ${c_source_end}
+        " PTHREAD_HAS_GETNAME)
+    if (PTHREAD_HAS_GETNAME)
+        target_compile_definitions(${target} PRIVATE -DAWS_PTHREAD_HAS_GETNAME)
```

Comment:
oh glorious, I googled pthread_setname_np and found docs pages for some less-common systems where it takes 2 args: [IBM i???](https://www.ibm.com/docs/en/i/7.2?topic=ssw_ibm_i_72/apis/pthread_getname_np.html) [MKS???](https://www.mkssoftware.com/docs/man3/pthread_getname_np.3.asp)

maybe stick with the _TAKES_3_ARGS naming scheme
your call if you want to support the _TAKES_2_ARGS case

Reply:
yeah. missed those 2. switched to 2/3 arg approach.